WTG.Analyzers/Analyzers/AwaitCompleted/AwaitCompletedAnalyzer.cs

```diff
@@ -0,0 +1,163 @@
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using WTG.Analyzers.Utils;
+
+namespace WTG.Analyzers
+{
+	[DiagnosticAnalyzer(LanguageNames.CSharp)]
+	public sealed partial class AwaitCompletedAnalyzer : DiagnosticAnalyzer
+	{
+		public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics { get; } = ImmutableArray.Create(
+			Rules.DontAwaitTriviallyCompletedTasksRule);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.RegisterCompilationStartAction(CompilationStart);
+		}
+
+		static void CompilationStart(CompilationStartAnalysisContext context)
+		{
+			var cache = new FileDetailCache();
+
+			context.RegisterSyntaxNodeAction(
+				c => Analyze(c, cache),
+				SyntaxKind.AwaitExpression);
+		}
+
+		static void Analyze(SyntaxNodeAnalysisContext context, FileDetailCache cache)
+		{
+			if (cache.IsGenerated(context.SemanticModel.SyntaxTree, context.CancellationToken))
+			{
+				return;
+			}
+
+			var node = (AwaitExpressionSyntax)context.Node;
+
+			if (IsAwaitTargetTrivial(context.SemanticModel, node, out var innerValueLocation, context.CancellationToken))
+			{
+				var additionalLocations = innerValueLocation == null
+					? Enumerable.Empty<Location>()
+					: new[] { innerValueLocation };
+
+				context.ReportDiagnostic(
+				Diagnostic.Create(
```

Comment:
Shouldn't this line be indented by an extra notch?

Reply:
Oops.